features/sharding/api/src/main/java/org/apache/shardingsphere/sharding/api/sharding/complex/ComplexKeysShardingValue.java

```diff
@@ -39,4 +40,6 @@
     private final Map<String, Collection<T>> columnNameAndShardingValuesMap;
     
     private final Map<String, Range<T>> columnNameAndRangeValuesMap;
+
+    private final DataNodeInfo dataNodeInfo;
```

Comment:
In ComplexKeysShardingValue, is the value of DataNodeInfo correct?

Reply:
Yes, I've already tested it

Reply:
Currently, DataNodeInfo is extracted by regular expression, which is suitable for auto table sharding algorithm, not user-defined sharding algorithm. I worry that using DataNodeInfo in ComplexKeysShardingValue may cause errors.

Reply:
So, can you provide some examples to show the use of DataNodeInfo in ComplexKeysShardingValue?

Reply:
I extended the name of the algorithm ：COMPLEX_LAST_MOD or COMPLEX_LAST_HASH_MOD